src/planning/nav2_planning_tests/planner_tester.cpp

```diff
@@ -0,0 +1,549 @@
+    char const * path = getenv("TEST_MAP");
+    if (path == NULL) {
+      file_path = "../../src/planning/nav2_planning_tests/maps/map.pgm";
+    } else {
+      file_path = std::string(path);
+    }
+  }
+
+  RCLCPP_INFO(this->get_logger(), "PlannerTester::loadMap: file_path: %s", file_path.c_str());
+
+  if (!yaml_file_name.empty()) {
+    // TODO(orduno): parse yaml file
+    RCLCPP_WARN(this->get_logger(), "PlannerTester::loadMap: yaml file parser not implemented"
+      " yet, will use default values instead");
+  }
+
+  try {
+    map_ =
+      std::make_shared<nav_msgs::msg::OccupancyGrid>(
+      map_loader::loadMapFromFile(
+        file_path, resolution, negate,
+        occupancy_threshold, free_threshold, origin, mode));
+  } catch (...) {
+    RCLCPP_ERROR(this->get_logger(),
+      "PlannerTester::loadDefaultMap: failed to load image from file: %s", file_path.c_str());
+    throw;
+  }
+
+  map_->header.stamp = this->now();
+  map_->header.frame_id = "map";
+  map_->info.map_load_time = this->now();
+
+  // TODO(orduno): replace with a latched topic
+  map_timer_ = create_wall_timer(1s, std::bind(&PlannerTester::mapCallback, this));
+
+  map_set_ = true;
+  costmap_set_ = false;
+  using_fake_costmap_ = false;
+
+  setCostmap();
+}
+
+void PlannerTester::mapCallback()
+{
+  map_publisher_->publish(map_);
+}
+
+void PlannerTester::setCostmap()
+{
+  RCLCPP_INFO(this->get_logger(), "PlannerTester::setCostmap");
+
+  if (!map_set_) {
+    RCLCPP_ERROR(
+      this->get_logger(),
+      "PlannerTester::setCostmap: map has not been provided");
+
+    // TODO(orduno) clarify how to handle errors here
+    return;
+  }
+
+  // TODO(orduno): clarify if we should be passing the node as shared or raw pointer
+  costmap_ = std::make_unique<Costmap>(
+    this, trinary_costmap_, track_unknown_space_, lethal_threshold_, unknown_cost_value_);
+
+  costmap_->setStaticMap(*map_);
+
+  costmap_set_ = true;
+  using_fake_costmap_ = false;
+}
+
+void PlannerTester::loadSimpleCostmap(const TestCostmap & testCostmapType)
+{
+  RCLCPP_INFO(this->get_logger(), "PlannerTester::loadSimpleCostmap");
+
+  if (costmap_set_) {
+    RCLCPP_WARN(
+      this->get_logger(),
+      "PlannerTester::loadSimpleCostmap: setting a new costmap with fake values");
+  }
+
+  costmap_ = std::make_unique<Costmap>(this);
+
+  costmap_->setTestCostmap(testCostmapType);
+
+  costmap_set_ = true;
+  using_fake_costmap_ = true;
+}
+
+void PlannerTester::startCostmapServer(std::string serviceName)
+{
+  RCLCPP_INFO(this->get_logger(), "PlannerTester::startCostmapServer");
+
+  if (!costmap_set_) {
+    RCLCPP_ERROR(
+      this->get_logger(),
+      "PlannerTester::startCostmapServer:: you need to set a costmap before starting the service");
+    return;
+  }
+
+  auto costmap_service_callback = [this](
+    const std::shared_ptr<rmw_request_id_t>/*request_header*/,
+    const std::shared_ptr<nav2_world_model_msgs::srv::GetCostmap::Request> request,
```

Comment:
We should start using typedefs and using directives. This is really awful to read.